BlogTemplate/Services/SlugGenerator.cs

```diff
@@ -20,11 +21,8 @@ public SlugGenerator(BlogDataStore dataStore)
         public string CreateSlug(string title)
         {
             string tempTitle = title;
-            char[] invalidChars = Path.GetInvalidFileNameChars();
-            foreach (char c in invalidChars)
-            {
-                tempTitle = tempTitle.Replace(c.ToString(), "");
-            }
+            Regex allowList = new Regex("([^A-Za-z0-9-])");

```

Comment:
I believe that the Regex type is thread safe for calling `.Replace(...)` multiple times, so you can declare just one `static` instance of this regular expression as a private field in this class, then just call `.Replace(...)` here.